spicy/toolchain/src/compiler/codegen/parsers/literals.cc

```diff
@@ -419,7 +421,8 @@ Expression* ParserBuilder::parseLiteral(const Production& p, Expression* dst) {
     if ( auto e = LiteralParser(this, &p, dst).buildParser(p.expression()) )
         return e;
 
-    hilti::logger().internalError(fmt("codegen: literal parser did not return expression for '%s'", *p.expression()));
+    hilti::logger().internalError(fmt("codegen: literal parser did not return expression for '%s' (%s)",
+                                      *p.expression(), p.expression()->typename_()));
```

Comment:
Just as a note and nothing to do here, printing the type is better than nothing, but TBH I'd probably still need to add logging to the code if I would need to figure out why this did not succeed (for the test this would print `expression::Ctor` which we have a handler above, but this only fails when recursing).

Reply:
Odd, I saw `hilti::ctor::Coerced`, that's how I found it.

Either way, this is not going to solve all problems, but it can be handy, have found it useful at other similar places already.